File: C/D2-반복문자 지우기.c
#include <stdio.h>

char stack[1000];
int top = -1;

void stack_init()
{
    for(int i = 0; i < 1000; i++)
        stack[i] = '\0';
    top = -1;
}

void push(char alpha)
{
    //우선 top을 증가시킨다
    top += 1;

    //overflow는 발생 안함
    //write
    stack[top] = alpha;
}

int pop()
{
    //underflow
    if(top == -1)
        return -1;

    //pop
    else
    {
        //pop할 때 지우는 기능 추가
        stack[top] == '\0';
        top -= 1;
        return 0;
    }
}

int len(char *str)
{   
    //문자열 길이 구하는 함수
    int length = 0;

    while(str[length] != '\0')
        length ++;

    return length;
}



int main()
{
    int T;
    scanf("%d", &T);

    for(int t = 0; t < T; t++)
    {

        //필요한 문자 받기
        char string[1000];
        scanf("%s ", &string);

        //문자열 길이 설정
        int length = len(string);

        //stack 초기화
        stack_init();

        for(int idx = 0; idx < length; idx++)
        {
            //이전에 push된 값과 동일하다면
            //이전 것도 pop시켜 지운다
            if(string[idx] == stack[top])
                pop();

            else
                push(string[idx]);
        }

        //결과 문자열 만들기
        int index = 0;

        while(stack[index] != '\0')
        {
            index ++;
        }
        
        printf("%s\n", stack);

        //결과 출력
        printf("#%d %d\n", t+1, index);
    }
    return 0;
}

File: Python/stack/D2-반복문자 지우기.py
stack = ['']*1000
top = -1

def my_push(alpha):
    
    global top

    #우선 top을 증가시킨다
    top += 1
    #overflow는 없을 것이다
    #write
    stack[top] = alpha

def my_pop():

    global top

    #underflow
    if(top == -1):
        return False
    
    #pop
    else:
        #pop할 때 지우는 값으로 ''할당
        stack[top] = ''
        top -= 1
        return stack[top+1]
        

T = int(input())

for t in range(1, T+1):

    #문자를 입력받아서 리스트로 만들기
    strlist = list(input())

    for idx in range(len(strlist)):

        #만약 이전에 push된 값과 동일하다면 
        #이전 것도 pop하고 지운다
        if(strlist[idx] == stack[top]):
            my_pop()

        else:
            my_push(strlist[idx])

    #결과 문자열 만들기
    index = 0
    result_str = ''

    while(stack[index] != ''):
        result_str += stack[index]
        index += 1

    #결과출력
    print(f'#{t} {len(result_str)}')

    #스택 초기화
    stack = ['']*1000
    top = -1